README.md

```diff
@@ -1,244 +1,90 @@
 [![](https://img.shields.io/pypi/v/lakefs-spec)](https://pypi.org/project/lakefs-spec) ![GitHub](https://img.shields.io/github/license/appliedAI-Initiative/lakefs-spec) [![docs](https://img.shields.io/badge/docs-latest-blue)](https://appliedai-initiative.github.io/lakefs-spec/)
  ![GitHub](https://img.shields.io/github/stars/appliedAI-Initiative/lakefs-spec)
 
-# lakefs-spec: An `fsspec` implementation for lakeFS
+# `lakefs-spec`: An `fsspec` backend for lakeFS
 
-This repository contains a [filesystem-spec](https://github.com/fsspec/filesystem_spec) implementation for the [lakeFS](https://lakefs.io/) project.
-Its main goal is to facilitate versioned data operations in lakeFS directly from Python code, for example using `pandas`. Data versioning enables reproducibility of experiments - a best practice in machine learning.
-
-See the examples below ([features](#usage), [versioning best-practices](#reproducibility-through-data-versioning-with-lakefs-and-lakefs-spec)) below for inspiration.
-
-A more detailed example is in the notebook in the [`/demos` directory](/demos/demo_data_science_project.ipynb).
+Welcome to `lakefs-spec`, a [filesystem-spec](https://github.com/fsspec/filesystem_spec) backend implementation for the [lakeFS](https://lakefs.io/) data lake.
+Our primary goal is to streamline versioned data operations in lakeFS, enabling seamless integration with popular data science tools such as pandas, polars, and DuckDB directly from Python.
 
 ## Installation
 
-To install the package directly from PyPI via `pip`, run
-
-```shell
-pip install --upgrade pip
-pip install lakefs-spec
-```
-
-or, for the bleeding edge version,
+`lakefs-spec` is published on PyPI, you can simply install it using your favorite package manager:
 
 ```shell
-pip install git+https://github.com/appliedAI-Initiative/lakefs-spec.git
+$ pip install lakefs-spec
+  # or
+$ poetry add lakefs-spec
 ```
 
-To add the project as a dependency using `poetry`, use
-
-```shell
-poetry add lakefs-spec
-```
-
-or, for the development version,
-
-```shell
-poetry add git+https://github.com/appliedAI-Initiative/lakefs-spec.git
-```
+If you want `lakefs-spec` to automatically discover and load credentials from an existing `~/.lakectl.yaml` credentials file on your machine, additionally install the `PyYAML` library.
```

Comment:
```suggestion
If you want `lakefs-spec` to automatically discover and load credentials from an existing `~/.lakectl.yaml` credentials file on your machine, additionally install the `PyYAML` library (`$ pip install pyyaml`).
```

Reply:
Maybe remove this line from the README, and add it on a page listing the different installation options?

Reply:
I was considering both of your suggestions but opted to put it in the readme because the examples under Usage require the YAML parsing (I left out the `pip install pyyaml`, since users by that point should hopefully know how to install a package, I don't want to repeat everything for both `pip` and `poetry` again 😅).

In general, I am leaning towards adding a dependency on `PyYAML` to the project dependencies to save our users some headaches. WDYT, @nicholasjng?

Reply:
I disagree - has any user reported any headaches? With all due respect, running `pip install --upgrade pyyaml` once in the console (which is even suggested if you fail to load an existing configfile) is possible for every prospective user of this library.

And since we only require it for an opt-in feature (lakectl configs), adding at as an unconditional requirement is not appropriate.

That being said, if we mention it heavily in the user guide, we can either a) change that to use other auth methods or b) put a small disclaimer in front of the examples (preferred).

Reply:
While I understand your reluctance to include a new mandatory dependency, I'd like to note that the lakectl config feature is not really opt-in, but rather the default behavior of the filesystem, unless the user customizes the constructor parameters.  If we consider it optional, we should change this behavior - if we want this to be the default, it should be as painless as possible for the user, IMO.

In any way, this PR isn't the place to a make this decision, so I would push this discussion into a follow-up issue.

Reply:
Resolved through #141